<launch>

<!-- Start Camera (point grey camera) -->
    <arg name="device" default = "17302081"/>
    <arg name="rate" default="30"/>

    <node name="single_reader" pkg="ptgrey_reader" type="single_reader"
         ns="pg_$(arg device)" output="log">
        <remap from="/image_out" to="image_raw" />
        <remap from="/image_out_roi" to="image" />
        <param name="is_pub" type="bool" value="true"/>
        <param name="is_show" type="bool" value="false"/>
        <param name="is_print" type="bool" value="false"/>
        <param name="serialNum" type="int" value="$(arg device)"/>
        <param name="frameRate" type="double" value="$(arg rate)"/>
        <param name="is_auto_shutter" type="bool" value="true"/>
        <param name="shutter" type="double" value="15.0"/>
        <param name="brightness" type="double" value="10.1"/>
        <param name="exposure" type="double" value="0.8"/>
        <param name="WB_red" type="int" value="550"/>
        <param name="WB_Blue" type="int" value="810"/>
        <param name="gain" type="double" value="5.0"/>
        <rosparam>
            is_sync:   false
            is_grey:   false
            is_roi:    true
            size_x:    1280
            size_y:    1024
            center_x:  640
            center_y:  512
            cropper_x: 1280
            cropper_y: 1024
            down_sample_scale: .5
        </rosparam>
    </node>



<!-- Start IMU
  Note: user need to be in dailout group. This is because the N3 is connected to
  i7 as a ttyUSB device. /dev/ttyUSB0. You either need to set the permissions
  so that the user can access it or best is to add user to group `dailout`.
-->
<node name="djiros" pkg="djiros" type="djiros_withtrigger_node" output="log" launch-prefix="">
    <param name="trigger_freq" type="int" value="15"/>
    <param name="log_level" type="string" value="info"/>
    <param name="serial_name" type="string" value="/dev/ttyUSB0"/>
    <param name="baud_rate" type="int" value="921600"/>
    <param name="app_id" type="int" value="$(env DJIROS_APPID)"/>
    <param name="enc_key" type="string" value="$(env DJIROS_ENCKEY)"/>
    <param name="app_version" type="int" value="1"/>
    <param name="sensor_mode" type="bool" value="false"/>
    <param name="align_with_fmu" type="bool" value="true"/>
</node>



<!-- Bag Recording -->
<node name="bag_record" pkg="rosbag" type="record" args="/pg_$(arg device)/image /djiros/imu " output="screen"/>


<!-- Viz -->
<node name="imshow" pkg="image_view" type="image_view" output="log" >
<remap from="image" to="/pg_$(arg device)/image" />
</node>

</launch>
